core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala

```diff
@@ -142,7 +144,83 @@ object ReassignPartitionsCommand extends Logging {
         TopicAndPartition(topic, partition) -> replicas
       }
     }
-    (partitionsToBeReassigned, currentAssignment)
+    (partitionsToBeReassigned, currentAssignment, disableRackAware match {
```

Comment:
I made a similar change. I also added a precondition for such balance reports: there need to be at least one rack that has 2+ more brokers than some other rack. This was missing in the previous patch.